app/src/hooks/useTokenPrice.tsx

```diff
@@ -0,0 +1,46 @@
+import { useQuery } from '@apollo/react-hooks'
+import { BigNumber, bigNumberify } from 'ethers/utils'
+import gql from 'graphql-tag'
+import { useEffect, useState } from 'react'
+
+import { getLogger } from '../util/logger'
+import { Status } from '../util/types'
+
+const logger = getLogger('useTokenPrice')
+
+const query = gql`
+  query GetTokenPrice($id: String!) {
+    token(id: $id) {
+      usdPerToken
+    }
+  }
+`
+
+type GraphResponse = {
+  token: {
+    usdPerToken: string
+  }
+}
+
+export const useTokenPrice = (address: string) => {
+  const [tokenPrice, setTokenPrice] = useState(0)
+
+  const { data: data, error: error, loading: loading } = useQuery<GraphResponse>(query, {
+    notifyOnNetworkStatusChange: true,
+    skip: false,
+    variables: { id: address.toLowerCase() },
+  })
+
+  useEffect(() => {
+    if (data && data.token.usdPerToken) {
+      setTokenPrice(Number(data.token.usdPerToken))
+    }
+  }, [data])
+
+  console.log(tokenPrice)
+
+  return {
+    tokenPrice,
+    status: error ? Status.Error : loading ? Status.Loading : Status.Ready,
```

Comment:
Small suggestion....When I was working on history table I found out that If its set like this it doesn't always display state correctly and when changed to this 
`status: error ? Status.Error : loading ? Status.Loading : data ? Status.Ready : Status.Loading,`
heres console logs with that implemented in this pr 
![Screenshot 2021-06-01 at 14 29 31](https://user-images.githubusercontent.com/33226956/120323761-20978680-c2e6-11eb-9689-4b303818f7ec.png)
And here is it like its now :
![Screenshot 2021-06-01 at 14 29 48](https://user-images.githubusercontent.com/33226956/120323809-3147fc80-c2e6-11eb-8254-2381ffd3ff58.png)
Although dunno if it causes some other issues when done like this?